### PR TITLE
Add version v0.1.0

### PR DESCRIPTION
**Pull request by gmelodie**
As discussed with @whyrusleeping, we should add a cutoff tag that people can refer back to since we'll be doing breaking changes and refators soon. This also adds a `--version/-v` flag to the estuary CLI.

TODO: `git push origin v0.1.0` after this is merged

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func init() {
 }
 
 var log = logging.Logger("estuary")
+var GitVersion string
```
Comment (alvin-reyes) [inline main.go:53]: Is the GitVersion different from the App version here? https://github.com/application-research/estuary/pull/242/files

Comment (gmelodie) [reply main.go:53]: Nope

**Review by alvin-reyes — APPROVED** [review]
LGTM

File: Makefile [hunk 2]
```diff
@@ -48,15 +47,16 @@ endif
...
 
-# Once estuary has it's own version cmd add this back in
-#ldflags=-X=github.com/application-research/estuary/version.GitVersion=$(GITVERSION)
-#ifneq ($(strip $(LDFLAGS)),)
-#	ldflags+=-extldflags=$(LDFLAGS)
-#endif
-#GOFLAGS+=-ldflags="$(ldflags)"
+# Add version information to the package
+# ldflags=-X=github.com/application-research/estuary/main.GitVersion=$(GITVERSION)
+ldflags=-X=main.GitVersion=$(GITVERSION)
```
Comment (en0ma) [inline Makefile:52]:
```suggestion
ldflags=-X=main.appVersion=$(VERSION)
```

File: Makefile [hunk 3]
```diff
@@ -48,15 +47,16 @@ endif
...
 CLEAN+=build/.update-modules
 
-# Once estuary has it's own version cmd add this back in
-#ldflags=-X=github.com/application-research/estuary/version.GitVersion=$(GITVERSION)
-#ifneq ($(strip $(LDFLAGS)),)
-#	ldflags+=-extldflags=$(LDFLAGS)
-#endif
-#GOFLAGS+=-ldflags="$(ldflags)"
+# Add version information to the package
+# ldflags=-X=github.com/application-research/estuary/main.GitVersion=$(GITVERSION)
```
Comment (en0ma) [inline Makefile:51]:
```suggestion
# ldflags=-X=github.com/application-research/estuary/main.appVersion=$(VERSION)
```

File: Makefile [hunk 4]
```diff
@@ -48,15 +47,16 @@ endif
...
+ldflags=-X=main.GitVersion=$(GITVERSION)
+ifneq ($(strip $(LDFLAGS)),)
+	ldflags+=-extldflags=$(LDFLAGS)
+endif
+GOFLAGS+=-ldflags="$(ldflags)"
 
 # using the commit for now as appVersion
-GOFLAGS=-ldflags "-X main.appVersion=$(COMMIT)"
+GOFLAGS=-ldflags "-X main.appVersion=$(VERSION)"
```
Comment (en0ma) [inline Makefile:59]: with my suggestion above, you can remove this line 58-59

**Review by en0ma — CHANGES_REQUESTED** [review]

**Review by en0ma — APPROVED** [review]

**Pull request merged**